torch/distributed/nn/jit/instantiator.py

```diff
@@ -0,0 +1,151 @@
+    ), "Expect forward in interface methods, while it has {}".format(
+        module_interface_c.getMethodNames()
+    )
+    method_schema = module_interface_c.getMethod("forward")
+
+    arg_str_list = []
+    arg_type_str_list = []
+    for argument in method_schema.arguments:
+        arg_str_list.append(argument.name)
+
+        if argument.has_default_value():
+            default_value_str = " = {}".format(argument.default)
+        else:
+            default_value_str = ""
+        arg_type_str = "{name}: {type}{default_value}".format(
+            name=argument.name, type=argument.type, default_value=default_value_str
+        )
+        arg_type_str_list.append(arg_type_str)
+
+    arg_str_list = arg_str_list[1:]  # Remove "self".
+    args_str = ", ".join(arg_str_list)
+
+    arg_type_str_list = arg_type_str_list[1:]  # Remove "self".
+    arg_types_str = ", ".join(arg_type_str_list)
+
+    assert len(method_schema.returns) == 1
+    argument = method_schema.returns[0]
+    return_type_str = str(argument.type)
+
+    return args_str, arg_types_str, return_type_str
+
+
+def _write(out_path, text):
+    try:
+        with open(out_path, "r") as f:
+            old_text = f.read()
+    except IOError:
+        old_text = None
+    if old_text != text:
+        with open(out_path, "w") as f:
+            print("Writing {}".format(out_path))
+            f.write(text)
+    else:
+        print("Skipped writing {}".format(out_path))
+
+
+def cleanup_generated_modules():
+    generated_module_paths = pathlib.Path(INSTANTIATED_TEMPLATE_DIR_PATH).glob("_remote_module*.py")
+    for file_path in generated_module_paths:
+        assert file_path.is_file(), f"Epect {file_path} to be a file"
+        print(f"Removing {file_path}")
+        file_path.unlink()
+
+
+def instantiate_remote_module_template(module_interface_cls, is_scriptable):
+    # Generate the template instance name.
+    module_interface_cls_name = torch.jit._qualified_name(module_interface_cls).replace(
+        ".", "_"
+    )
+    generated_module_name = f"_remote_module_{module_interface_cls_name}"
+
+    if is_scriptable:
+        args_str, arg_types_str, return_type_str = get_arg_return_types_from_interface(
+            module_interface_cls
+        )
+        kwargs_str = ""
+        arrow_and_return_type_str = f" -> {return_type_str}"
+        arrow_and_future_return_type_str = f" -> Future[{return_type_str}]"
+    else:
+        args_str = "*args"
+        kwargs_str = "**kwargs"
+        arg_types_str = "*args, **kwargs"
+        arrow_and_return_type_str = ""
+        arrow_and_future_return_type_str = ""
+
+    jit_decorator_str_map = dict(
+        jit_script_decorator="@torch.jit.script",
+        jit_export_decorator="@torch.jit.export",
+        jit_interface_decorator="@torch.jit.interface",
+    )
+    if is_scriptable is False:
+        jit_decorator_str_map = {
+            key: "" for key, value in jit_decorator_str_map.items()
+        }
```

Comment:
move `jit_decorator_str_map` before the above if-else, and merge this if into the `else` clause.

Reply:
I have split the scripted and non scripted code paths in latest version.